ducktape/services/service.py

```diff
@@ -181,10 +184,11 @@ def free(self):
             node.account.logger = None
             self.free_node(node)
 
+        self.nodes = []
+
     def free_node(self, node):
```

Comment:
Can we get rid of `free_node`? It's not safe to call on its own -- it looks like `free` is supposed to be the only safe way to clean up nodes now. At a minimum, seems like this should be `_free_node`.